hack/homebrew/update-homebrew-package.sh

```diff
@@ -72,7 +76,7 @@ mv tanzu-community-edition-updated.rb tanzu-community-edition.rb
 
 git add tanzu-community-edition.rb
 
-git commit -m "auto-generated - update tce homebrew formula for version ${version}"
+git commit -s -m "auto-generated - update tce homebrew formula for version ${version}"
```

Comment:
Please see above

Reply:
@dvonthenen About the `--admin` flag, I see `gh pr merge` mentioning this -

```
      --admin            Use administrator privileges to merge a pull request that does not meet requirements
      --auto             Automatically merge only after necessary requirements are met
```

And we want to run checks and merge the PR only after the GitHub checks pass. For Homebrew package update PR, the GitHub checks are here - https://github.com/vmware-tanzu/homebrew-tanzu/blob/main/.github/workflows/check-tce-homebrew-formula-pr.yaml

If the checks fail we want the PR to not be merged. The checks ensure that the formula works by doing a install test currently. Ideally an upgrade test should also happen as part of the checks, or we need to do the tests before we raise the PR but that's not how we do it currently

In order to run the checks and then auto merge the PR we use `--auto` and avoided using `--admin` which merges the PR without the checks passing

https://github.com/vmware-tanzu/community-edition/blob/0e10a503a4098ae45b8a1b59d80598759fac40e0/hack/homebrew/update-homebrew-package.sh#L85

Reply:
Going by the restrictions on auto merging to main, I guess we need `--admin` and hence cannot do checks in PR, so I guess we can do checks / tests before we raise PR and then fail if checks fail, that way no PR is also created, which is also better, that way we don't have PRs that don't work. If the checks pass, we raise the PR and use `--admin` to merge the PR

Reply:
> The checks ensure that the formula works by doing a install test currently.
Hii @karuppiah7890 for this I am modifying the update version script and testing before raising the PR.

Reply:
for brew script we are using `--admin` as we are running a check before merging in the script itself.
for choco are moving with `--auto` as we have to run a PR check.

Reply:
as long as the change isn't significant, `--admin` should be ok. most of the automation only bumps tiny things like version numbers. to your excellent point though, if something does go horribly wrong, creating a PR gives us an easy button to revert the commit in the PR quickly.

if `--auto` works for your change/implementation, will leave it up to you on what works best for your case. just know that it will add some delay to merge.

Reply:
If https://github.com/vmware-tanzu/community-edition/pull/3242 got success will use `--admin` for choco too.